src/main/java/com/mraof/minestuck/block/ItemBlockPair.java

```diff
@@ -0,0 +1,44 @@
+package com.mraof.minestuck.block;
+
+import com.mraof.minestuck.item.MSItems;
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.level.ItemLike;
+import net.minecraft.world.level.block.Block;
+import net.minecraftforge.registries.RegistryObject;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@MethodsReturnNonnullByDefault
+public record ItemBlockPair<B extends Block, I extends Item>(RegistryObject<B> blockHolder, RegistryObject<I> itemHolder) implements ItemLike
```

Comment:
The name is practically self explanatory but it still seems wise to add some javadoc. At least to explain if there are any exceptions to usage, peculiarities when referencing an instance, etc